internal/history/service.go

```diff
@@ -223,8 +205,56 @@ func (_ *evaluationHistoryService) ListEvaluationHistory(
 		slices.Reverse(rows)
 	}
 
+	propsvc := propertiessvc.NewPropertiesService(qtx)
+
+	data := make([]OneEvalHistoryAndEntity, 0, len(rows))
+	for _, row := range rows {
+		ent, err := qtx.GetEntityByID(ctx, db.GetEntityByIDParams{
+			ID:       row.EntityID,
+			Projects: []uuid.UUID{row.ProjectID},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error getting entity: %w", err)
+		}
+
+		p, err := ehs.providerManager.InstantiateFromID(ctx, ent.ProviderID)
+		if err != nil {
+			return nil, fmt.Errorf("error instantiating provider: %w", err)
+		}
+
+		propClient, err := provifv1.As[provifv1.Provider](p)
```

Comment:
I don't think you need to cast to Provider here, what Instantiate returns is already a Provider, right?